### ISSUE TITLE
Support for Tailwind

### ISSUE THREAD

**Issue by Wazarr94** [streamsync-cloud/streamsync#15]
I have taken a look at the project and it's quite mind-blowing. it sure has a lot of potential and I will be looking to explore it over the next weeks.

Is it possible to add support for [TailwindCSS](https://tailwindcss.com/) styling? It's one of the most popular tools for styling in the web dev world and something that I think could be integrated into the visual editor quite well.

**Comment by ramedina86** [comment]
I've decided to move forward with this, it'll be added to the roadmap as likely 2nd-3rd priority. I'll work on publishing a public roadmap so that everyone can be aware.

It'll be a great addition for more advanced use cases with minimal disruption to those wanting the basic experience.

**Comment by Wazarr94** [comment]
Very nice.
I've spent a good time reading the code, it shouldn't be too hard to add support for Tailwind in the HTML Element component in the first place. And the rest of components will follow, if necessary.
I'll try to file a PR tomorrow if all goes well.

**Comment by ramedina86** [comment]
I think the idea is great in general but there are a few design considerations and will probably take some iterations to nail it.

Perhaps for the moment you can focus on the following? Would be helpful

- Identifying npm package, what's the best version, is there a minified one
- Can we dynamically import to reduce the load size if Tailwind isn't being used?
- To check whether to import, how can we detect if Tailwind is being used? / Where are we storing this data?
- How can we deliver the best Tailwind experience? Autocomplete? Something like chips would be nice 
- How do we actually apply these styles in a general way? It'd probably involve ComponentProxy

I like an expert/advanced switch of some sort, as you suggested, but I'm not sold on the dropdown. I'd like to do a few Figma mocks to see how to best fit this in terms of UX. There's also the consideration of the current CSS of _HTML Element_ and how to integrate it to the new way.

I'm thinking something like 

GENERAL
- Text
- Alignment
- etc
STYLE
- Color
- Separator
- Custom styling
    ( Tailwind )     ( CSS )           - imagine these are grey chips like the "Default"/"CSS"/"Pick"
   [rounded-full X ]   [mx-auto X]        - these would be square chips with a X (remove) button, perhaps overkill and an input text will do

In summary, if you don't mind, please explore the more technical part first and I'll get back to you in terms of UX

**Comment by Wazarr94** [comment]
I've done a first implementation for HTML Element here: [GitHub repository](https://github.com/Wazarr94/streamsync/tree/tailwind)

![image](https://user-images.githubusercontent.com/39807740/236628196-a0d6249c-0204-4882-aaed-f7708ee990fd.png)

I've used the latest tailwind version. The only problem I faced was that since no tailwind is used in the vue code, by default Tailwind remove all of its classes. I used a plugin to force tailwind to import some classes. I made a list which is incomplete of course, open to suggestions. We can't dynamically import Tailwind, but the css file doesn't get too big so I don't see much of an issue. The static directory was around 23mb before I used Tailwind and kept the same size after.

Normally, the styles you have used for your elements should override the tailwind styles (I'm thinking about the container class) so there shouldn't be a need for adding a prefix in Tailwind. In case I'm wrong, it is not hard to add a prefix in my code.

About the tailwind experience, I think chips would probably be hard to implement given the number of tailwind classes. I'm pretty sure a text input is ok.

From my experience, there is a need to give tailwind styling to other elements, such as text, button or layout elements. It can be complicated since your elements have many layers. PrimeVue managed to deal with this problem with a passthrough object: [PrimeVue button](https://primevue.org/button/#pt.doc.button).

There could be a new `class` property in the ComponentProxy which is an object that looks like the one they use. The builder interface would show the style for each element in the component.
